### DISCUSSION TITLE
Add support for React Server Components (RSC) (Next.js 13 app dir)

### DISCUSSION THREAD

**Discussion by ghost** [rainbow-me/rainbowkit#1219] [Ideas]
I would like for Rainbow Kit to support [Next.js 13.4](https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration) with app directory using React Server Components (RSC).

If I try to use "@rainbow-me/rainbowkit": "^1.0.0" with "next": "^13.4.1", using the app directory I get the following error:

```
./src/app/page.tsx
ReactServerComponentsError:

You're importing a component that needs useState. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.

      ,-[/rainbow/node_modules/.pnpm/@rainbow-me+rainbowkit@1.0.0_@types+react@18.2.6_react-dom@18.2.0_react@18.2.0_viem@0.3.21_wagmi@1.0.2/node_modules/@rainbow-me/rainbowkit/dist/chunk-BZODBG62.js:4085:1]
 4085 | }
 4086 | 
 4087 | // src/components/ConnectOptions/MobileOptions.tsx
 4088 | import React50, { useCallback as useCallback8, useContext as useContext14, useState as useState10 } from "react";
      :                                                                            ^^^^^^^^
 4089 | 
 4090 | // src/components/ConnectOptions/MobileOptions.css.ts
 4091 | var scroll = "_1am14410";
      `----

The error was caused by importing '@rainbow-me/rainbowkit/dist/index.js' in './src/app/page.tsx'.

Maybe one of these should be marked as a client entry with "use client":
  ./src/app/page.tsx
```

I have already added "use client" to the required files in [my rainbowkit fork](https://github.com/ramozdev/rainbowkit/commit/3713cf5869b9cc4c54b697cd9472c20122e6c6e0). I ask for your permission to submit a pull request.

**Comment by DanielSinclair** [comment]
@ramozdev RainbowKit and Wagmi are purely client-side libraries, so with the App Router they will only work if you tag the top of your layout file with `use client;` as well the file where you have imported the RainbowKit `ConnectButton`
